app/controllers/merchant/items_controller.rb

```diff
@@ -8,6 +8,23 @@ def new
     @merchant = Merchant.find(current_user[:merchant_id])
   end
 
+  def edit
+    @item = Item.find(params[:id])
+  end
+
+  def update
+    @item = Item.find(params[:id])
+    if @item.update(item_params)
+      @item.save
+      flash[:notice] = "Item Succesfully Updated"
+      redirect_to "/merchant/items"
+    else
+      flash[:error] = "Incorrectly filled out #{@item.changed_attributes.keys.join(", ")}, try again."
+      @item.restore_attributes
```

Comment:
Why would we need this? If update fails to save shouldn’t it revert back to the previous self?